tests/e2e/specs/blocks/content-helper/filters.spec.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies.
+ */
+import {
+	enablePageDialogAccept,
+} from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	insertRecordIntoTaxonomy,
+	getContentHelperMessage,
+	setUserDisplayName,
+	startUpTest,
+	setSiteKeys,
+} from '../../../utils';
+
+/**
+ * Tests for the Content Helper filters.
+ */
+describe( 'Content Helper filters', () => {
+	/**
+	 * Prevents browser from locking with dialogs, logs in to WordPress,
+	 * activates the Parse.ly plugin, and sets valid site keys.
+	 */
+	beforeAll( async () => {
+		enablePageDialogAccept();
+		await startUpTest();
+		await setSiteKeys( 'blog.parsely.com', 'test' );
+	} );
+
+	/**
+	 * Verifies that the Content Helper attempts to fetch results when a Site ID
+	 * and API Secret are provided.
+	 */
+	it( 'Should attempt to fetch results when a Site ID and API Secret are provided', async () => {
+		await setUserDisplayName( 'admin', '' );
+
+		expect( await getContentHelperMessage() ).toMatch( 'The Parse.ly API did not return any results for top-performing posts by the author "admin".' );
+	} );
+
+	/**
+	 * Verifies that the Content Helper respects the author > category > tag
+	 * filter prioritization order, with author being the weakest and tag the
+	 * strongest.
+	 *
+	 * Note: This test inserts the category/tag into the database before
+	 * selecting it in the WordPress Post Editor.
+	 */
+	it( 'Should be prioritized in the correct order', async () => {
+		const firstName = 'Andrew';
+		const lastName = 'Montalenti';
+		const categoryName = 'Parse.ly Tech';
+		const tagName = 'changelog';
+
+		await setUserDisplayName( firstName, lastName );
+		await insertRecordIntoTaxonomy( categoryName, 'category' );
+		await insertRecordIntoTaxonomy( tagName, 'post_tag' );
+
+		// Author.
+		expect( await getContentHelperMessage() ).toMatch( 'Top-performing posts by the author "' + firstName + ' ' + lastName + '".' );
+
+		// Author + category.
+		expect( await getContentHelperMessage( categoryName ) ).toMatch( 'Top-performing posts in the category "' + categoryName + '".' );
+
+		// Author + tag.
+		expect( await getContentHelperMessage( null, tagName ) ).toMatch( 'Top-performing posts with the tag "' + tagName + '".' );
+
+		// Author + category + tag.
+		expect( await getContentHelperMessage( categoryName, tagName ) ).toMatch( 'Top-performing posts with the tag "' + tagName + '".' );
```

Comment:
Personal preference, but I'd find these easier to read with template strings.
```suggestion
		expect( await getContentHelperMessage() ).toMatch( `Top-performing posts by the author "${ firstName } ${ lastName }".` );

		// Author + category.
		expect( await getContentHelperMessage( categoryName ) ).toMatch( `Top-performing posts in the category "${ categoryName }".` );

		// Author + tag.
		expect( await getContentHelperMessage( null, tagName ) ).toMatch( `Top-performing posts with the tag "${ tagName }".` );

		// Author + category + tag.
		expect( await getContentHelperMessage( categoryName, tagName ) ).toMatch( `Top-performing posts with the tag "${ tagName }".` );
```